examples/pytorch/mnist-ddp/mnist.ipynb

```diff
@@ -0,0 +1,627 @@
+      "Train Epoch: 1 [55000/60000 (92%)]\tLoss: 0.356350\n",
+      "Train Epoch: 1 [56000/60000 (93%)]\tLoss: 0.363343\n",
+      "Train Epoch: 1 [57000/60000 (95%)]\tLoss: 0.482411\n",
+      "Train Epoch: 1 [58000/60000 (97%)]\tLoss: 0.353806\n",
+      "Train Epoch: 1 [59000/60000 (98%)]\tLoss: 0.274754\n",
+      "Training is finished\n"
+     ]
+    }
+   ],
+   "source": [
+    "import os\n",
+    "\n",
+    "# Set the Torch Distributed env variables so the training function can be run in the notebook\n",
+    "# See https://pytorch.org/docs/stable/elastic/run.html#environment-variables\n",
+    "os.environ[\"RANK\"] = \"0\"\n",
+    "os.environ[\"LOCAL_RANK\"] = \"0\"\n",
+    "os.environ[\"WORLD_SIZE\"] = \"1\"\n",
+    "os.environ[\"MASTER_ADDR\"] = \"localhost\"\n",
+    "os.environ[\"MASTER_PORT\"] = \"1234\"\n",
+    "\n",
+    "# Run the training function locally\n",
+    "train_fashion_mnist({\"epochs\": 1})"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create the Kubeflow Training Client"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "from kubeflow.training import Trainer, TrainingClient\n",
+    "client = TrainingClient()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## List the training runtimes"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Runtime(name='torch-distributed', phase='pre-training', accelerator='Unknown', accelerator_count='Unknown')\n"
+     ]
+    }
+   ],
+   "source": [
+    "for runtime in client.list_runtimes():\n",
+    "    print(runtime)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Run the distributed training Job"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 49,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "job_name = client.train(\n",
+    "    # Use one the of the training runtimes installed on your Kubernetes cluster\n",
+    "    runtime_ref=\"torch-distributed\",\n",
+    "    trainer=Trainer(\n",
+    "        func=train_fashion_mnist,\n",
+    "        func_args={\n",
+    "            \"epochs\": 10,\n",
+    "        },\n",
+    "        # Set how many worker Pods you want the job to be distributed into\n",
+    "        num_nodes=4,\n",
+    "        # Set the resources for each worker Pod\n",
+    "        resources_per_node={\n",
+    "            \"cpu\": 8,\n",
+    "            \"memory\": \"16Gi\",\n",
+    "            \"nvidia.com/gpu\": 1,\n",
+    "        },\n",
+    "    ),\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Check the training job components"
+   ]
```

Comment:
Please also explain what does it mean:
```
Since we created TrainJob using 4 nodes,  TrainJob will create 4 components: `trainer-node-0`, `trainer-node-1`. ....
```